x-pack/elastic-agent/pkg/core/monitoring/config/config.go

```diff
@@ -6,9 +6,19 @@ package config
 
 // MonitoringConfig describes a configuration of a monitoring
 type MonitoringConfig struct {
-	Enabled        bool `yaml:"enabled" config:"enabled"`
-	MonitorLogs    bool `yaml:"logs" config:"logs"`
-	MonitorMetrics bool `yaml:"metrics" config:"metrics"`
+	Enabled        bool                  `yaml:"enabled" config:"enabled"`
+	MonitorLogs    bool                  `yaml:"logs" config:"logs"`
+	MonitorMetrics bool                  `yaml:"metrics" config:"metrics"`
+	HTTP           *MonitoringHTTPConfig `yaml:"http" config:"http"`
+}
+
+// MonitoringHTTPConfig is a config defining HTTP endpoint published by agent
+// for other processes to watch its metrics.
+// Processes are only exposed when HTTP is enabled.
+type MonitoringHTTPConfig struct {
+	Enabled bool   `yaml:"enabled" config:"enabled"`
+	Host    string `yaml:"host" config:"host"`
+	Port    int    `yaml:"port" config:"port"`
```

Comment:
Let's add a 'positive' validator (see [unpack](https://pkg.go.dev/github.com/elastic/go-ucfg#Config.Unpack)) docs. Then if port is configured, but empty we will fail to parse the configuration and fail with the setting that failed. The `Enabled` will be used to not start the server.